### ISSUE TITLE
Hundreds of experiments started throwing HfHubHTTPError: 504 Server Error: Gateway Time-out for url Error

### ISSUE THREAD

**Issue by AntreasAntoniou** [huggingface/huggingface_hub#1516]
### Describe the bug

I have been using huggingface models, as a place to keep some private models that I am working on. Just now, all my experiments began complaining that they couldn't access the HFHub Gateway, with errors such as

```bash
HfHubHTTPError: 504 Server Error: Gateway Time-out for URL  https://huggingface.co/api/models/Antreas/timm-ade20k-debug-137/commit/main
```

### Reproduction

Simply running the following causes the error:

```bash
import huggingface_hub


def create_hf_model_repo(hf_repo_path: str) -> str:
    huggingface_hub.login(
        token=os.environ["HF_TOKEN"], add_to_git_credential=True
    )
    print(f"Creating repo {hf_repo_path}")
    return huggingface_hub.create_repo(
        hf_repo_path, repo_type="model", exist_ok=True, private=True
    )
```

### Logs

```shell
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /opt/conda/envs/main/lib/python3.10/site-packages/huggingface_hub/utils/_errors.py:259 in        │
│ hf_raise_for_status                                                                              │
│                                                                                                  │
│   256 │   </Tip>                                                                                 │
│   257 │   """                                                                                    │
│   258 │   try:                                                                                   │
│ ❱ 259 │   │   response.raise_for_status()                                                        │
│   260 │   except HTTPError as e:                                                                 │
│   261 │   │   error_code = response.headers.get("X-Error-Code")                                  │
│   262                                                                                            │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/requests/models.py:1021 in raise_for_status    │
│                                                                                                  │
│   1018 │   │   │   )                                                                             │
│   1019 │   │                                                                                     │
│   1020 │   │   if http_error_msg:                                                                │
│ ❱ 1021 │   │   │   raise HTTPError(http_error_msg, response=self)                                │
│   1022 │                                                                                         │
│   1023 │   def close(self):                                                                      │
│   1024 │   │   """Releases the connection back to the pool. Once this method has been            │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
HTTPError: 504 Server Error: Gateway Time-out for url: https://huggingface.co/api/models/Antreas/timm-ade20k-debug-138/commit/main

The above exception was the direct cause of the following exception:

╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /devcode/GATE/gate/run.py:209 in <module>                                                        │
│                                                                                                  │
│   206                                                                                            │
│   207                                                                                            │
│   208 if __name__ == "__main__":                                                                 │
│ ❱ 209 │   run()                                                                                  │
│   210                                                                                            │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/hydra/main.py:94 in decorated_main             │
│                                                                                                  │
│    91 │   │   │   │   else:                                                                      │
│    92 │   │   │   │   │   # no return value from run_hydra() as it may sometime actually run t   │
│    93 │   │   │   │   │   # multiple times (--multirun)                                          │
│ ❱  94 │   │   │   │   │   _run_hydra(                                                            │
│    95 │   │   │   │   │   │   args=args,                                                         │
│    96 │   │   │   │   │   │   args_parser=args_parser,                                           │
│    97 │   │   │   │   │   │   task_function=task_function,                                       │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/hydra/_internal/utils.py:394 in _run_hydra     │
│                                                                                                  │
│   391 │   │                                                                                      │
│   392 │   │   if args.run or args.multirun:                                                      │
│   393 │   │   │   run_mode = hydra.get_mode(config_name=config_name, overrides=overrides)        │
│ ❱ 394 │   │   │   _run_app(                                                                      │
│   395 │   │   │   │   run=args.run,                                                              │
│   396 │   │   │   │   multirun=args.multirun,                                                    │
│   397 │   │   │   │   mode=run_mode,                                                             │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/hydra/_internal/utils.py:457 in _run_app       │
│                                                                                                  │
│   454 │   │   │   overrides.extend(["hydra.mode=MULTIRUN"])                                      │
│   455 │                                                                                          │
│   456 │   if mode == RunMode.RUN:                                                                │
│ ❱ 457 │   │   run_and_report(                                                                    │
│   458 │   │   │   lambda: hydra.run(                                                             │
│   459 │   │   │   │   config_name=config_name,                                                   │
│   460 │   │   │   │   task_function=task_function,                                               │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/hydra/_internal/utils.py:223 in run_and_report │
│                                                                                                  │
│   220 │   │   return func()                                                                      │
│   221 │   except Exception as ex:                                                                │
│   222 │   │   if _is_env_set("HYDRA_FULL_ERROR") or is_under_debugger():                         │
│ ❱ 223 │   │   │   raise ex                                                                       │
│   224 │   │   else:                                                                              │
│   225 │   │   │   try:                                                                           │
│   226 │   │   │   │   if isinstance(ex, CompactHydraException):                                  │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/hydra/_internal/utils.py:220 in run_and_report │
│                                                                                                  │
│   217                                                                                            │
│   218 def run_and_report(func: Any) -> Any:                                                      │
│   219 │   try:                                                                                   │
│ ❱ 220 │   │   return func()                                                                      │
│   221 │   except Exception as ex:                                                                │
│   222 │   │   if _is_env_set("HYDRA_FULL_ERROR") or is_under_debugger():                         │
│   223 │   │   │   raise ex                                                                       │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/hydra/_internal/utils.py:458 in <lambda>       │
│                                                                                                  │
│   455 │                                                                                          │
│   456 │   if mode == RunMode.RUN:                                                                │
│   457 │   │   run_and_report(                                                                    │
│ ❱ 458 │   │   │   lambda: hydra.run(                                                             │
│   459 │   │   │   │   config_name=config_name,                                                   │
│   460 │   │   │   │   task_function=task_function,                                               │
│   461 │   │   │   │   overrides=overrides,                                                       │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/hydra/_internal/hydra.py:132 in run            │
│                                                                                                  │
│   129 │   │   callbacks.on_run_end(config=cfg, config_name=config_name, job_return=ret)          │
│   130 │   │                                                                                      │
│   131 │   │   # access the result to trigger an exception in case the job failed.                │
│ ❱ 132 │   │   _ = ret.return_value                                                               │
│   133 │   │                                                                                      │
│   134 │   │   return ret                                                                         │
│   135                                                                                            │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/hydra/core/utils.py:260 in return_value        │
│                                                                                                  │
│   257 │   │   │   sys.stderr.write(                                                              │
│   258 │   │   │   │   f"Error executing job with overrides: {self.overrides}" + os.linesep       │
│   259 │   │   │   )                                                                              │
│ ❱ 260 │   │   │   raise self._return_value                                                       │
│   261 │                                                                                          │
│   262 │   @return_value.setter                                                                   │
│   263 │   def return_value(self, value: Any) -> None:                                            │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/hydra/core/utils.py:186 in run_job             │
│                                                                                                  │
│   183 │   │   with env_override(hydra_cfg.hydra.job.env_set):                                    │
│   184 │   │   │   callbacks.on_job_start(config=config, task_function=task_function)             │
│   185 │   │   │   try:                                                                           │
│ ❱ 186 │   │   │   │   ret.return_value = task_function(task_cfg)                                 │
│   187 │   │   │   │   ret.status = JobStatus.COMPLETED                                           │
│   188 │   │   │   except Exception as e:                                                         │
│   189 │   │   │   │   ret.return_value = e                                                       │
│                                                                                                  │
│ /devcode/GATE/gate/run.py:118 in run                                                             │
│                                                                                                  │
│   115 │   # Set the seed for reproducibility                                                     │
│   116 │   set_seed(seed=cfg.seed)                                                                │
│   117 │                                                                                          │
│ ❱ 118 │   ckpt_dict = create_hf_model_repo_and_download_maybe(                                   │
│   119 │   │   cfg=cfg,                                                                           │
│   120 │   │   hf_repo_path=cfg.hf_repo_path,                                                     │
│   121 │   │   hf_cache_dir=cfg.hf_cache_dir,                                                     │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/gate/boilerplate/utils.py:415 in               │
│ create_hf_model_repo_and_download_maybe                                                          │
│                                                                                                  │
│   412 ) -> Tuple[Optional[pathlib.Path], str]:                                                   │
│   413 │   create_hf_model_repo(hf_repo_path)                                                     │
│   414 │   create_directories(hf_cache_dir)                                                       │
│ ❱ 415 │   upload_config_files(                                                                   │
│   416 │   │   cfg=cfg, hf_repo_path=hf_repo_path, hf_cache_dir=hf_cache_dir                      │
│   417 │   )                                                                                      │
│   418 │   checkpoint_store_dir = (                                                               │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/gate/boilerplate/utils.py:369 in               │
│ upload_config_files                                                                              │
│                                                                                                  │
│   366 │   for filepath, path_in_repo in [                                                        │
│   367 │   │   (config_yaml_path, "config.yaml"),                                                 │
│   368 │   ]:                                                                                     │
│ ❱ 369 │   │   hf_api.upload_file(                                                                │
│   370 │   │   │   repo_id=hf_repo_path,                                                          │
│   371 │   │   │   path_or_fileobj=filepath.as_posix(),                                           │
│   372 │   │   │   path_in_repo=path_in_repo,                                                     │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py:118 in    │
│ _inner_fn                                                                                        │
│                                                                                                  │
│   115 │   │   if check_use_auth_token:                                                           │
│   116 │   │   │   kwargs = smoothly_deprecate_use_auth_token(fn_name=fn.__name__, has_token=ha   │
│   117 │   │                                                                                      │
│ ❱ 118 │   │   return fn(*args, **kwargs)                                                         │
│   119 │                                                                                          │
│   120 │   return _inner_fn  # type: ignore                                                       │
│   121                                                                                            │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/huggingface_hub/hf_api.py:826 in _inner        │
│                                                                                                  │
│    823 │   │   │   return self.run_as_future(fn, self, *args, **kwargs)                          │
│    824 │   │                                                                                     │
│    825 │   │   # Otherwise, call the function normally                                           │
│ ❱  826 │   │   return fn(self, *args, **kwargs)                                                  │
│    827 │                                                                                         │
│    828 │   _inner.is_future_compatible = True  # type: ignore                                    │
│    829 │   return _inner  # type: ignore                                                         │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/huggingface_hub/hf_api.py:3205 in upload_file  │
│                                                                                                  │
│   3202 │   │   │   path_in_repo=path_in_repo,                                                    │
│   3203 │   │   )                                                                                 │
│   3204 │   │                                                                                     │
│ ❱ 3205 │   │   commit_info = self.create_commit(                                                 │
│   3206 │   │   │   repo_id=repo_id,                                                              │
│   3207 │   │   │   repo_type=repo_type,                                                          │
│   3208 │   │   │   operations=[operation],                                                       │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py:118 in    │
│ _inner_fn                                                                                        │
│                                                                                                  │
│   115 │   │   if check_use_auth_token:                                                           │
│   116 │   │   │   kwargs = smoothly_deprecate_use_auth_token(fn_name=fn.__name__, has_token=ha   │
│   117 │   │                                                                                      │
│ ❱ 118 │   │   return fn(*args, **kwargs)                                                         │
│   119 │                                                                                          │
│   120 │   return _inner_fn  # type: ignore                                                       │
│   121                                                                                            │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/huggingface_hub/hf_api.py:826 in _inner        │
│                                                                                                  │
│    823 │   │   │   return self.run_as_future(fn, self, *args, **kwargs)                          │
│    824 │   │                                                                                     │
│    825 │   │   # Otherwise, call the function normally                                           │
│ ❱  826 │   │   return fn(self, *args, **kwargs)                                                  │
│    827 │                                                                                         │
│    828 │   _inner.is_future_compatible = True  # type: ignore                                    │
│    829 │   return _inner  # type: ignore                                                         │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/huggingface_hub/hf_api.py:2712 in              │
│ create_commit                                                                                    │
│                                                                                                  │
│   2709 │   │                                                                                     │
│   2710 │   │   try:                                                                              │
│   2711 │   │   │   commit_resp = get_session().post(url=commit_url, headers=headers, data=data,  │
│ ❱ 2712 │   │   │   hf_raise_for_status(commit_resp, endpoint_name="commit")                      │
│   2713 │   │   except RepositoryNotFoundError as e:                                              │
│   2714 │   │   │   e.append_to_message(_CREATE_COMMIT_NO_REPO_ERROR_MESSAGE)                     │
│   2715 │   │   │   raise                                                                         │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/huggingface_hub/utils/_errors.py:301 in        │
│ hf_raise_for_status                                                                              │
│                                                                                                  │
│   298 │   │                                                                                      │
│   299 │   │   # Convert `HTTPError` into a `HfHubHTTPError` to display request information       │
│   300 │   │   # as well (request id and/or server error message)                                 │
│ ❱ 301 │   │   raise HfHubHTTPError(str(e), response=response) from e                             │
│   302                                                                                            │
│   303                                                                                            │
│   304 def _format_error_message(message: str, request_id: Optional[str], server_message: Optio   │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
HfHubHTTPError: 504 Server Error: Gateway Time-out for url: https://huggingface.co/api/models/Antreas/timm-ade20k-debug-138/commit/main
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /opt/conda/envs/main/bin/accelerate:8 in <module>                                                │
│                                                                                                  │
│   5 from accelerate.commands.accelerate_cli import main                                          │
│   6 if __name__ == '__main__':                                                                   │
│   7 │   sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])                         │
│ ❱ 8 │   sys.exit(main())                                                                         │
│   9                                                                                              │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/accelerate/commands/accelerate_cli.py:45 in    │
│ main                                                                                             │
│                                                                                                  │
│   42 │   │   exit(1)                                                                             │
│   43 │                                                                                           │
│   44 │   # Run                                                                                   │
│ ❱ 45 │   args.func(args)                                                                         │
│   46                                                                                             │
│   47                                                                                             │
│   48 if __name__ == "__main__":                                                                  │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/accelerate/commands/launch.py:941 in           │
│ launch_command                                                                                   │
│                                                                                                  │
│   938 │   elif defaults is not None and defaults.compute_environment == ComputeEnvironment.AMA   │
│   939 │   │   sagemaker_launcher(defaults, args)                                                 │
│   940 │   else:                                                                                  │
│ ❱ 941 │   │   simple_launcher(args)                                                              │
│   942                                                                                            │
│   943                                                                                            │
│   944 def main():                                                                                │
│                                                                                                  │
│ /opt/conda/envs/main/lib/python3.10/site-packages/accelerate/commands/launch.py:603 in           │
│ simple_launcher                                                                                  │
│                                                                                                  │
│   600 │   process.wait()                                                                         │
│   601 │   if process.returncode != 0:                                                            │
│   602 │   │   if not args.quiet:                                                                 │
│ ❱ 603 │   │   │   raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)    │
│   604 │   │   else:                                                                              │
│   605 │   │   │   sys.exit(1)                                                                    │
│   606                                                                                            │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
CalledProcessError: Command '['/opt/conda/envs/main/bin/python', 'gate/run.py', 'exp_name=timm-ade20k-debug-138', 'model=timm-segmentation-transformer', 'dataset=ade20k', 
'trainer=image_semantic_segmentation', 'evaluator=image_semantic_segmentation', 'seed=2306', 'train_batch_size=8', 'eval_batch_size=8', 'num_workers=48', 'optimizer.lr=0.00006', 
'model.timm_model_name=vit_base_patch16_clip_224.laion2b', 'optimizer.weight_decay=0.01', 'train_iters=100000']' returned non-zero exit status 1.
```

### System info

```shell
- huggingface_hub version: 0.15.1
- Platform: Linux-5.19.0-38-generic-x86_64-with-glibc2.35
- Python version: 3.10.11
- Running in iPython ?: No
- Running in notebook ?: No
- Running in Google Colab ?: No
- Token path ?: /root/.cache/huggingface/token
- Has saved token ?: True
- Who am I ?: Antreas
- Configured git credential helpers: !f()
- FastAI: N/A
- Tensorflow: N/A
- Torch: 2.0.1+cu118
- Jinja2: 3.1.2
- Graphviz: N/A
- Pydot: N/A
- Pillow: 9.3.0
- hf_transfer: N/A
- gradio: N/A
- numpy: 1.24.1
- ENDPOINT: https://huggingface.co
- HUGGINGFACE_HUB_CACHE: /root/.cache/huggingface/hub
- HUGGINGFACE_ASSETS_CACHE: /root/.cache/huggingface/assets
- HF_TOKEN_PATH: /root/.cache/huggingface/token
- HF_HUB_OFFLINE: False
- HF_HUB_DISABLE_TELEMETRY: False
- HF_HUB_DISABLE_PROGRESS_BARS: None
- HF_HUB_DISABLE_SYMLINKS_WARNING: False
- HF_HUB_DISABLE_EXPERIMENTAL_WARNING: False
- HF_HUB_DISABLE_IMPLICIT_TOKEN: False
- HF_HUB_ENABLE_HF_TRANSFER: False
```

**Comment by AntreasAntoniou** [comment]
Worth noting that the code that is now failing was battle tested over a period of 8 months now. So I don't think the code on my end is causing the issue, unless any of the methods I note above, or their usage have been deprecated or modified.

**Comment by narugo1992** [comment]
I also encountered the exact same issue. Multiple huggingface APIs (including repository and commit creation) are throwing a 504 error. I tried different network environments, but the issue persists. Additionally, accessing the huggingface website and uploading files using LFS are normal. However, manually creating a commit on the website (e.g., deleting files) also results in a 504 error. 

Another noteworthy observation is that although the HTTP response is 504, it seems that all the mentioned operations are actually being executed. After receiving the 504 error, the commit I attempted to create appears in the repository approximately 3 minutes later, with a creation time of "3 minutes ago".
